### PR TITLE
licensing: implement cody landing page

### LINKED ISSUE #59693: Implement Cody landing page

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes #59693

On a cody-only license, enterprise users won't have access to Cody, so we provide them with a landing page for Cody to upsell the product.

CleanShot 2024-01-30 at 21 48 18

## Test plan

* Start up a Sourcegraph instance with a Code Search only license.
* Navigate to the home page
* Clicking on the `Cody Ai` link on the Navbar should show a landing page for upselling Cody.

### CODE REVIEW

**Review by camdencheek — APPROVED** [review]
I did not give a close review of the upsell page, but given that the code there is very isolated to that page, as long as it passes visual inspection I think we're good. Looks great -- thanks for all your work on this!

File: client/web/src/cody/upsell/CodyUpsellPage.module.scss [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+.no-access-banner {
+    background-color: white;
+    padding: 4rem;
+    margin: 0 auto;
+    margin-top: -5%;
+    margin-bottom: 4.5rem;
+    width: 70%;
+    text-align: center;
+    border-radius: 1rem;
+    border: 1px solid var(--gray-04);
```
Comment (taiyab) [inline client/web/src/cody/upsell/CodyUpsellPage.module.scss]: @BolajiOlajide The colors here shouldn't generally use the root colors (`gray-01`, `gray-02`, `gray-03` `...`) but rather the semantic equivalents.

File: client/web/src/cody/upsell/CodyUpsellPage.module.scss [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        padding: 0;
+        color: var(--gray-08);
+    }
+}
+
+.cody-intro {
+    width: 70%;
+    height: fit-content;
+    margin: 0 auto;
+    border-radius: 1rem;
```
Comment (taiyab) [inline client/web/src/cody/upsell/CodyUpsellPage.module.scss]: Can we can reduce down the `border-radius` on all the elements with `1rem`  to `0.5rem`?

**Review by fkling — APPROVED** [review]
👍🏻

File: client/web/src/cody/upsell/CodyUpsellPage.module.scss (hunk 2 above)
Comment (BolajiOlajide) [reply client/web/src/cody/upsell/CodyUpsellPage.module.scss]: Yup. Pushed that.

File: client/web/src/cody/upsell/CodyUpsellPage.module.scss (hunk 1 above)
Comment (BolajiOlajide) [reply client/web/src/cody/upsell/CodyUpsellPage.module.scss]: Yeah, I missed out on this one. Thanks.

Pushed a fix for this.

**Pull request merged**